Adding examples to spam doc of akismet/recaptcha configs to add to staticman.yml

RyanTG opened the pull request:
Happy to revise this, based on your review.

RyanTG committed:
Adding examples to spam doc of akismet/recaptcha configs to add to staticman.yml

alexwaibel commented on docs/spam.md:
````diff
@@ -23,15 +23,37 @@ Another effective method of combating spam is to use reCAPTCHA. If you enable Ak
   <input type="hidden" name="options[reCaptcha][secret]" value="{ENCRYPTED-SECRET}">
   ```
-1. Follow the reCAPTCHA documentation to add either a <a href="https://developers.google.com/recaptcha/docs/display#auto_render" class="cta">checkbox</a> or <a href="https://developers.google.com/recaptcha/docs/invisible#auto_render" class="cta">invisible challenge</a> to your form
+1. Add the reCAPTCHA credentials to your staticman.yml
+  ``` yml
+    reCaptcha:
+      enabled: true
+      # siteKey and secret should match your config.toml
+      siteKey: ""
+      # (!) ENCRYPT reCaptcha secret key using Staticman /encrypt endpoint
+      # i.e. https://{your-staticman-api}/v3/encrypt/{your-site-secret}
+      # For more information, visit https://staticman.net/docs/encryption
+      secret: ""
+  ```
````
This is the same as the current step 2:
> 2. Use the Staticman encryption endpoint to encrypt the secret and add the reCAPTCHA options to your site config

Can move this example yaml up so it's right under that step?

alexwaibel replied:
Also, which config is `config.toml` referring to?

RyanTG replied:
I believe `config.toml` a hugo-specific reference. I removed it. And the other comment about encrypting the secret was redundant with the text in step 2, so I removed that too.

I haven't implemented reCAPTCHA on my own jekyll/staticman sites, so I'm not sure if these instructions could be further clarified.

alexwaibel commented on docs/spam.md:
````diff
@@ -23,15 +23,37 @@ Another effective method of combating spam is to use reCAPTCHA. If you enable Ak
 
 1. Sign up for an <a href="https://akismet.com/" class="cta">Akismet account</a>. Note your Akismet API key and give Akismet your static site's URL
 1. Add the Akismet API key and URL to the <a href="/docs/api" class="cta">API config</a>
+1. Add the Akismet config to your staticman.yml file, making sure that the values below match your form's fields
+  ``` yml
+  # Akismet spam detection
+    akismet:
+      enabled: true
+      author: "name"
+      authorEmail: "email"
+      authorUrl: "website"
+      content: "message"
+  ```
````
```suggestion
1. Add the Akismet config to your staticman.yml site config file, making sure that the values below match the name properties of your form's fields
  ``` yml
    akismet:
      enabled: true
      author: "name"
      authorEmail: "email"
      authorUrl: "website"
      content: "message"
  ```
```
nit

RyanTG replied:
Oops, one minute. I missed this review note.

alexwaibel reviewed (changes requested):
Thank you for submitting this. I agree that inline examples here would be helpful. I've left some small suggestions.

RyanTG committed:
Clean up spam doc based on review

RyanTG committed:
further clarified the akismet/recaptcha instructions

RyanTG commented:
I believe I addressed everything. I also fixed a link to the encryption instructions.

I struggled a bit to get the code snippets to be correctly-formatted. 1) They look kinda ugly on the page, because there's an extra border being added (at least in Firefox), and 2) The indentation in the first line is off. It's like it really wants to force an indentation on the first line. In the code you can clearly see that I do not want to indent the first line. But I think it's good enough for now.

alexwaibel commented:
Thanks again for submitting this and for the speedy revision. The Firefox double borders on code blocks are something I've seen as well but just haven't had time to dig into. I agree they're ugly but at least all they seem to function fine otherwise. I'll open an issue for this to keep track of it.
